contracts/StealthRelayer.sol

```diff
@@ -27,12 +28,26 @@ contract StealthRelayer is Governable, CollectableDust, StealthTx, IStealthRelay
     _;
   }
 
+  modifier setsCaller() {
+    caller = msg.sender;
+    _;
+    caller = address(0);
+  }
+
   function execute(
     address _job,
     bytes memory _callData,
     bytes32 _stealthHash,
     uint256 _blockNumber
-  ) external payable override onlyValidJob(_job) validateStealthTxAndBlock(_stealthHash, _blockNumber) returns (bytes memory _returnData) {
+  )
+    external
+    payable
+    override
+    onlyValidJob(_job)
+    validateStealthTxAndBlock(_stealthHash, _blockNumber)
+    setsCaller()
```

Comment:
I think `setsCaller` needs to be the first one to be called (before all the `validate...()`, right ? If we want to use `caller()` from outside to do any validations.